src/python/xml_parser-py.c

```diff
@@ -91,16 +88,24 @@ c_pkgcb(cr_Package *pkg,
     PyObject *arglist, *result, *py_pkg;
     CbData *data = cbdata;
 
-    if (data->py_pkg)
-        py_pkg = data->py_pkg;
-    else
-        py_pkg = Object_FromPackage(pkg, 1);
+    PyObject *keyFromPtr = PyLong_FromVoidPtr(pkg);
+    py_pkg = PyDict_GetItem(data->py_pkgs, keyFromPtr);
+    if (py_pkg) {
+        arglist = Py_BuildValue("(O)", py_pkg);
+        result = PyObject_CallObject(data->py_pkgcb, arglist);
+        Py_DECREF(arglist);
+        PyDict_DelItem(data->py_pkgs, keyFromPtr);
+    } else {
+        // The package was not provided by user in c_newpkgcb,
+        // create new python package object
+        PyObject *new_py_pkg = Object_FromPackage(pkg, 1);
+        arglist = Py_BuildValue("(O)", new_py_pkg);
+        result = PyObject_CallObject(data->py_pkgcb, arglist);
+        Py_DECREF(arglist);
```

Comment:
`arglist` could be decref'd once at the end rather than on both branches.

Reply:
I will move the `Py_DECREF` after that branch, thanks. :slightly_smiling_face: